internal/logger/default_logger.go

```diff
@@ -21,124 +21,114 @@ import (
 	"log"
 	"os"
 	"runtime"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-go-client/logger"
 )
 
 const (
 	// logCallDepth is used for removing the last two method names from call trace when logging file names.
 	logCallDepth    = 3
-	defaultLogLevel = infoLevel
+	defaultLogLevel = logger.InfoLevel
 	tracePrefix     = "TRACE"
 	warnPrefix      = "WARN"
 	debugPrefix     = "DEBUG"
 	errorPrefix     = "ERROR"
 	infoPrefix      = "INFO"
 )
 
-// DefaultLogger has Go's built in log embedded in it. It adds level logging.
+// DefaultLogger has Go's built-in logger embedded in it. It adds level logging.
 // To set the logging level, one should use the LoggingLevel property. For example
 // to set it to debug level:
 //  config.SetProperty(property.LoggingLevel.Name(), logger.DebugLevel)
 // If loggerConfig.SetLogger() method is called, the LoggingLevel property will not be used.
 type DefaultLogger struct {
 	*log.Logger
-	Level int
+	Weight logger.Weight
 }
 
 // New returns a Default Logger with defaultLogLevel.
 func New() *DefaultLogger {
-	return &DefaultLogger{
-		Logger: log.New(os.Stderr, "", log.LstdFlags),
-		Level:  defaultLogLevel,
-	}
+	l, _ := NewWithLevel(defaultLogLevel) // defaultLogLevel exists, no err
+	return l
 }
 
-func NewWithLevel(loggingLevel int) *DefaultLogger {
+func NewWithLevel(loggingLevel logger.Level) (*DefaultLogger, error) {
+	numericLevel, err := logger.GetLogLevel(loggingLevel)
+	if err != nil {
+		return nil, err
+	}
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
-		Level:  loggingLevel,
-	}
-}
-
-// Debug logs the given arguments at debug level if the level is greater than or equal to debug level.
-func (l *DefaultLogger) Debug(f func() string) {
-	if l.CanLogDebug() && f != nil {
-		s := fmt.Sprintf("DEBUG: %s", f())
-		l.Output(logCallDepth, s)
-	}
-}
-
-func (l *DefaultLogger) Trace(f func() string) {
-	if l.canLogTrace() && f != nil {
-		s := fmt.Sprintf("TRACE: %s", f())
-		l.Output(logCallDepth, s)
-	}
-}
-
-// Info logs the given arguments at info level if the level is greater than or equal to info level.
-func (l *DefaultLogger) Info(args ...interface{}) {
-	if l.canLogInfo() {
-		callerName := l.findCallerFuncName()
-		s := callerName + "\n" + infoPrefix + ": " + fmt.Sprint(args...)
-		l.Output(logCallDepth, s)
-	}
+		Weight: numericLevel,
+	}, nil
 }
 
-func (l *DefaultLogger) Infof(format string, values ...interface{}) {
-	if l.canLogInfo() {
-		s := fmt.Sprintf("INFO : %s", fmt.Sprintf(format, values...))
-		l.Output(logCallDepth, s)
+func (l *DefaultLogger) Log(wantedLevel logger.Weight, formatter func() string) {
+	if l.Weight < wantedLevel {
```

Comment:
Isn't this line is wrong ? 
Lets say that we called 
```
// Infof formats the given string with the given values, if logger level is info or finer.
func (la LogAdaptor) Infof(format string, values ...interface{}) {
	la.Log(logger.WeightInfo, func() string {
		return fmt.Sprintf(format, values...)
	})
}
```
WeightInfo is 400. 
And lets say that the user configured to see the everything which means he set WeightTrace which means 
l.weight is Trace which is 600.

This means that we will not log in this case. But it should have been logged. Right ?